src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -77,6 +78,16 @@ export const gatewayTransactionsReducer = handleActions<GatewayTransactionsState
             // pushing a newer transaction to the existing list messes the transactions order
             // this happens when most recent transactions are added to the existing txs in the store
             newHistory[startOfDate] = newHistory[startOfDate].sort((a, b) => b.timestamp - a.timestamp)
+
+            // remove tx from queued if it exists there
+            if (isMultisigExecutionInfo(transaction.executionInfo)) {
```

Comment:
It could be 'any' type of transaction.

Reply:
The executionInfo has type `ExecutionInfo = ModuleExecutionInfo | MultisigExecutionInfo` . In this approach I need to access the tx nonce to remove it from the `queued` queue and ModuleExecutionInfo does not have a nonce property.

Reply:
Not all transactions have a nonce. It is best to check the `safeTxHash` to cover all types.